lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1125,6 +1096,11 @@ def run(self):
         return msg, changed, rebuilt, differences.get_legacy_docker_diffs(), facts
 
 
+def _detect_publish_mode_usage(client):
+    for publish_def in client.module.params['publish']:
+        return 'mode' in publish_def
```

Comment:
```suggestion
        return bool(publish_def.get('mode'))
```
That should be compatible to `if port.get('mode'):` above.

Reply:
Of course forgot about that one.